src/psyclone/tests/psyir/transformations/transformations_test.py

```diff
@@ -444,6 +443,135 @@ def create_loops(body):
+                real, dimension(10, 10, 10) :: zwt, zwd, zwi, zws
+                scalar1 = 1
+                do jk = 2, jpkm1, 1
+                  do jj = 2, jpjm1, 1
+                    do ji = 2, jpim1, 1
+                       if (.true.) then
+                          scalar1 = zwt(ji,jj,jk)
+                       endif
+                       scalar2 = scalar1 + zwt(ji,jj,jk)
+                       zws(ji,jj,jk) = scalar2
+                    enddo
+                  enddo
+                enddo
+            end subroutine
+        end module my_mod''')
+    omplooptrans = OMPParallelLoopTrans()
+    loop = psyir.walk(Loop)[0]
+    omplooptrans.apply(loop)
+    expected = '''\
+    !$omp parallel do default(shared), private(ji,jj,jk,scalar2), \
+firstprivate(scalar1), schedule(auto)
+    do jk = 2, jpkm1, 1
+      do jj = 2, jpjm1, 1
+        do ji = 2, jpim1, 1
+          if (.true.) then
+            scalar1 = zwt(ji,jj,jk)
+          end if
+          scalar2 = scalar1 + zwt(ji,jj,jk)
+          zws(ji,jj,jk) = scalar2
+        enddo
+      enddo
+    enddo
+    !$omp end parallel do\n'''
+
+    gen = fortran_writer(psyir)
+    assert expected in gen
+    assert Compile(tmpdir).string_compiles(gen)
+
+    # Example with a read before write
+    psyir = fortran_reader.psyir_from_source('''
+        module my_mod
+            contains
+            subroutine my_subroutine()
+                integer :: ji, jj, jk, jpkm1, jpjm1, jpim1, scalar1, scalar2
+                real, dimension(10, 10, 10) :: zwt, zwd, zwi, zws
+                do jk = 2, jpkm1, 1
+                  do jj = 2, jpjm1, 1
+                    do ji = 2, jpim1, 1
+                       scalar2 = scalar1 + zwt(ji,jj,jk)
+                       scalar1 = 3
+                       zws(ji,jj,jk) = scalar2 + scalar1
+                    enddo
+                  enddo
+                enddo
+            end subroutine
+        end module my_mod''')
+    omplooptrans = OMPParallelLoopTrans()
+    loop = psyir.walk(Loop)[0]
+    # This need to be forced since the DependencyAnalysis wrongly considers
+    # it a reduction
+    omplooptrans.apply(loop, options={"force": True})
+    expected = '''\
+    !$omp parallel do default(shared), private(ji,jj,jk,scalar2), \
+firstprivate(scalar1), schedule(auto)
+    do jk = 2, jpkm1, 1
+      do jj = 2, jpjm1, 1
+        do ji = 2, jpim1, 1
+          scalar2 = scalar1 + zwt(ji,jj,jk)
+          scalar1 = 3
+          zws(ji,jj,jk) = scalar2 + scalar1
+        enddo
+      enddo
+    enddo
+    !$omp end parallel do'''
+
+    gen = fortran_writer(psyir)
+    assert expected in gen
+    assert Compile(tmpdir).string_compiles(gen)
+
+
+def test_omplooptrans_apply_firstprivate_fail(fortran_reader, fortran_writer):
+    ''' Test applying the OMPLoopTrans in cases where a firstprivate
+    clause it is needed to generate functionally equivalent code than
+    the starting serial version.
+    
```

Comment:
Trailing whitespace and unused `fortran_writer` argument.

Reply:
Removed